### PR TITLE
Add new indexing parameter and update performance tuning instruction

### LINKED ISSUE #8482: [DOC] Vector Search now supports new advanced setting to improve indexing performance
**What do you want to do?**
 
- [x] Request a change to existing documentation
- [ ] Add new documentation
- [ ] Report a technical problem with the documentation
- [ ] Other 

**Tell us about your request.** Provide a summary of the request.
In 2.18 we will be releasing a new advanced setting where users can configure to decide when to build graph. We would like to update [this](https://opensearch.org/docs/latest/search-plugins/knn/performance-tuning/) page to add new section 

***Version:** List the OpenSearch version to which this issue applies, e.g. 2.14, 2.12--2.14, or all.
2.18
 
**What other resources are available?** Provide links to related issues, POCs, steps for testing, etc.
https://github.com/opensearch-project/k-NN/issues/1942

### PR DESCRIPTION
**Pull request by VijayanB**
### Description
Add new indexing parameter and update performance tuning instruction

### Issues Resolved
Closes #8482  

### Version
2.18

### Frontend features
_If you're submitting documentation for an OpenSearch Dashboards feature, add a video that shows how a user will interact with the UI step by step. A voiceover is optional._ 

### Checklist
- [x] By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license and subject to the [Developers Certificate of Origin](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: _search-plugins/knn/knn-index.md [hunk 1]
```diff
@@ -368,6 +368,7 @@ Setting | Default | Updatable | Description
 :--- | :--- | :--- | :---
 `index.knn` | false | false | Whether the index should build native library indexes for the `knn_vector` fields. If set to false, the `knn_vector` fields will be stored in doc values, but approximate k-NN search functionality will be disabled.
 `index.knn.algo_param.ef_search` | 100 | true | The size of the dynamic list used during k-NN searches. Higher values result in more accurate but slower searches. Only available for NMSLIB.
+`index.knn.advanced.approximate_threshold` | 15000   | true      | This threshold defines the minimum number of vectors required in segment, before creating specialized vector data structures for approximate search. Set `-1` to disable building vector data structures and `0` to build always.
```
Comment (navneet1v) [inline _search-plugins/knn/knn-index.md:371]: should we mention somewhere that this setting change will only be applied to indices >= 2.18 version of Opensearch?

Comment (VijayanB) [reply _search-plugins/knn/knn-index.md:371]: This will be merged in 2.18 and will be available going forward ( following versions)

Comment (navneet1v) [reply _search-plugins/knn/knn-index.md:371]: So you are saying we are relying of the documentation version to expose this information. make sense.

**Review by navneet1v — APPROVED** [review]

**Comment by VijayanB** [comment]
@kolchfa-aws Can i get review on this? Thanks

**Comment by kolchfa-aws** [comment]
@VijayanB I rewrote and added example requests. Please look the changes over and let me know if I should change anything. Then we'll move the PR to editorial review. Thank you!

**Comment by VijayanB** [comment]
@kolchfa-aws Looks good to me. Thanks for making additional changes.

**Review by kolchfa-aws — APPROVED** [review]
LGTM. Thank you, @VijayanB!

File: _search-plugins/knn/knn-index.md [hunk 2]
```diff
@@ -368,6 +368,7 @@ Setting | Default | Updatable | Description
 :--- | :--- | :--- | :---
 `index.knn` | false | false | Whether the index should build native library indexes for the `knn_vector` fields. If set to false, the `knn_vector` fields will be stored in doc values, but approximate k-NN search functionality will be disabled.
 `index.knn.algo_param.ef_search` | 100 | true | The size of the dynamic list used during k-NN searches. Higher values result in more accurate but slower searches. Only available for NMSLIB.
+`index.knn.advanced.approximate_threshold` | 15,000   | true      | The number of vectors in a segment that must be reached before creating specialized data structures for approximate search. Set to `-1` to disable building vector data structures and `0` to always build them.
```
Comment (natebower) [inline _search-plugins/knn/knn-index.md:371]: "The number of vectors that a segment must have before..."?

File: _search-plugins/knn/performance-tuning.md [hunk 3]
```diff
@@ -98,6 +98,57 @@ In OpenSearch 2.15 or later, you can further improve indexing speed and reduce d
 This is an expert-level setting. Disabling the `_recovery_source` may lead to failures during peer-to-peer recovery. Before disabling the `_recovery_source`, check with your OpenSearch cluster admin to determine whether your cluster performs regular flushes before starting the peer-to-peer recovery of shards before disabling the `_recovery_source`.
```
Comment (natebower) [inline _search-plugins/knn/performance-tuning.md:98]:
```suggestion
This is an expert-level setting. Disabling the `_recovery_source` may lead to failures during peer-to-peer recovery. Before disabling the `_recovery_source`, check with your OpenSearch cluster admin to determine whether your cluster performs regular flushes before starting the peer-to-peer recovery of shards prior to disabling the `_recovery_source`.  
```

File: _search-plugins/knn/performance-tuning.md [hunk 4]
```diff
@@ -98,6 +98,57 @@ In OpenSearch 2.15 or later, you can further improve indexing speed and reduce d
 This is an expert-level setting. Disabling the `_recovery_source` may lead to failures during peer-to-peer recovery. Before disabling the `_recovery_source`, check with your OpenSearch cluster admin to determine whether your cluster performs regular flushes before starting the peer-to-peer recovery of shards before disabling the `_recovery_source`.  
 {: .warning}
 
+### (Expert-level) Build vector data structures on demand
```
Comment (natebower) [inline _search-plugins/knn/performance-tuning.md:101]:
```suggestion
### (Expert level) Build vector data structures on demand
```

File: _search-plugins/knn/performance-tuning.md [hunk 5]
```diff
@@ -98,6 +98,57 @@ In OpenSearch 2.15 or later, you can further improve indexing speed and reduce d
...
+
+This approach is recommended only for workloads that involve a single initial bulk upload and will be used exclusively for search after force merging to a single segment.
+
+During indexing, vector search builds a specialized data structure for a `knn_vector` field to enable efficient approximate k-NN search. However, these structures are rebuilt from scratch during [force merge]({{site.url}}{{site.baseurl}}/api-reference/index-apis/force-merge/) for k-NN indexes. To optimize indexing speed, follow these steps:
```
Comment (natebower) [inline _search-plugins/knn/performance-tuning.md:105]: "from scratch" is idiomatic. Can we just remove it or replace it with something like "totally rebuilt"? Additionally, following that, "during [force merge] operations on k-NN indexes"?

File: _search-plugins/knn/performance-tuning.md [hunk 6]
```diff
@@ -98,6 +98,57 @@ In OpenSearch 2.15 or later, you can further improve indexing speed and reduce d
...
+
+1. **Disable vector data structure creation**: Disable building vector data structures for new segments by setting [`index.knn.advanced.approximate_threshold`]({{site.url}}{{site.baseurl}}/search-plugins/knn/knn-index/#index-settings) to `-1` either at index creation or at any time after that:
```
Comment (natebower) [inline _search-plugins/knn/performance-tuning.md:107]:
```suggestion
1. **Disable vector data structure creation**: Disable vector data structure creation for new segments by setting [`index.knn.advanced.approximate_threshold`]({{site.url}}{{site.baseurl}}/search-plugins/knn/knn-index/#index-settings) to `-1` either at or after index creation:
```

File: _search-plugins/knn/performance-tuning.md [hunk 7]
```diff
@@ -98,6 +98,57 @@ In OpenSearch 2.15 or later, you can further improve indexing speed and reduce d
...
+
+    ```json
+    PUT /test-index/_settings
+    {
+      "index.knn.advanced.approximate_threshold": "0"
+    }
+    ```
+    {% include copy-curl.html %}
+
+    If you forget to reset the setting to `0` before the force merge, you will need to reindex your data.
```
Comment (natebower) [inline _search-plugins/knn/performance-tuning.md:140]:
```suggestion
    If you do not reset the setting to `0` before the force merge, you will need to reindex your data.
```

File: _search-plugins/knn/performance-tuning.md [hunk 8]
```diff
@@ -98,6 +98,57 @@ In OpenSearch 2.15 or later, you can further improve indexing speed and reduce d
...
+    {: .note}
+
+1. **Force merge segments into one segment**: Perform a force merge and specify `max_num_segments=1` to create the vector data structures only once:
+
+    ```json
+    POST test-index/_forcemerge?max_num_segments=1
+    ```
+    {% include copy-curl.html %}
+
+    After the force merge, with vector data structures built, new search requests will execute approximate k-NN search.
```
Comment (natebower) [inline _search-plugins/knn/performance-tuning.md:150]: "After the force merge, new search requests will execute approximate k-NN search using the newly created data structures"?

**Review by natebower — COMMENTED** [review]
@kolchfa-aws @VijayanB Please see my comments and changes and let me know if you have any questions. Thanks!

File: _search-plugins/knn/performance-tuning.md (hunk 8 above)
Comment (kolchfa-aws) [inline _search-plugins/knn/performance-tuning.md:150]:
```suggestion
    After the force merge, new search requests will execute approximate k-NN search using the newly created data structures.
```

File: _search-plugins/knn/performance-tuning.md (hunk 5 above)
Comment (kolchfa-aws) [inline _search-plugins/knn/performance-tuning.md:105]:
```suggestion
During indexing, vector search builds a specialized data structure for a `knn_vector` field to enable efficient approximate k-NN search. However, these structures are rebuilt during [force merge]({{site.url}}{{site.baseurl}}/api-reference/index-apis/force-merge/) for k-NN indexes. To optimize indexing speed, follow these steps:
```

Comment (kolchfa-aws) [reply _search-plugins/knn/performance-tuning.md]: Reworded.

**Pull request merged**